[ Site Isolation ] Modify macOS site-isolation post-commit bots to run without cross-origin--frame

robert-jenner opened the pull request:
#### f37cb327b07ac124741910bc857201a1cfcf4d70
<pre>
[ Site Isolation ] Modify macOS site-isolation post-commit bots to run without cross-origin--frame
<a href="https://rdar.apple.com/164042144">rdar://164042144</a>

Reviewed by Jonathan Bedard and Sihui Liu.

Modifying exisitng site-isolation framework to run without cross-origin-iframe. Also, adding API-tests to site-isolation test runs.

* LayoutTests/platform/mac-site-isolation/TestExpectations:
* Tools/CISupport/build-webkit-org/config.json:
* Tools/CISupport/build-webkit-org/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/build-webkit-org/steps.py:
(RunAPITests):
* Tools/Scripts/webkitpy/layout_tests/controllers/single_test_runner.py:
(SingleTestRunner):
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
(parse_args):
(_set_up_derived_options):

Canonical link: <a href="https://commits.webkit.org/302686@main">https://commits.webkit.org/302686@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ce1a10b4f6b59840b3f8b73b29fd5a96612d533a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/129912 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/2173 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/40770 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/137305 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/81402 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/db631552-00c3-41fb-98a2-8f25a3593d11) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/131783 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/2127 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/2064 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/98958 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/81402 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/132859 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/1597 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/116351 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/79651 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/0bb46e9e-2c21-4724-bf68-85a9c392af7a) 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/129262 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/1508 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/34479 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/80575 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/110016 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/34987 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/139786 "Built successfully") | | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/153/builds/1967 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/1823 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/107464 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/129342 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/2012 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/112699 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/107351 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27325 "Built successfully and passed tests") | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/164/builds/1575 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/31171 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/54769 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/2040 "Built successfully") | | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/157/builds/1854 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/163/builds/1889 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/1963 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

ryanhaddad commented on Tools/CISupport/build-webkit-org/config.json:
```diff
@@ -230,7 +230,7 @@
                   },
                   { "name": "Apple-Tahoe-Release-WK2-Site-Isolation-Tree-Tests", "factory": "TestLayoutFactory",
```
I think you'll need to change the factory to `TestLayoutAndAPIOnlyFactory` if you want to add API tests.

robert-jenner replied:
Thank you!! I'll go look at that.

robert-jenner replied:
Fixed it. Thank you so much!

szewai commented on Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
```diff
@@ -357,7 +357,7 @@ def parse_args(args):
         optparse.make_option(
             "--site-isolation", action="store_true", default=False,
-            help=("Run each test in a cross origin iframe with and without site isolation enabled and compare the results. Uses site-isolation test expectations")),
+            help=("Run each test with site isolation enabled. Uses site-isolation test expectations")),
```
As discussed on Slack, let's change it to "Run each test with and without site isolation enabled and compare the results."

JonWBedard replied:
Do we have examples of folks using this type of workflow? While I agree that's a desirable feature, I'm not aware of folks using this type of workflow in `run-webkit-tests`. It feels out-of-scope for this change.

JonWBedard replied:
@robert-jenner found https://commits.webkit.org/243771@main (which humorously, I reviewed), is that the sort of workflow you're referring to?

robert-jenner replied:
@szewai I have made the requested change here.

szewai commented on Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
```diff
@@ -488,9 +488,7 @@ def _set_up_derived_options(port, options):
-        if not options.load_in_cross_origin_iframe:
-            _log.warning("Option --site-isolation will set --load-in-cross-origin-iframe")
-        options.load_in_cross_origin_iframe = True
+        options.additional_header = 'SiteIsolationEnabled=true'
```
The changes needed are to remove `if options.site_isolation:` here, and remove ` runInCrossOriginFrame=true` in Tools/Scripts/webkitpy/layout_tests/controllers/single_test_runner.py.

robert-jenner replied:
I have made your requested changes to this PR.

robert-jenner committed:
[ Site Isolation ] Modify macOS site-isolation post-commit bots to run without cross-origin--frame
rdar://164042144

Reviewed by Jonathan Bedard and Sihui Liu.

Modifying exisitng site-isolation framework to run without cross-origin-iframe. Also, adding API-tests to site-isolation test runs.

* LayoutTests/platform/mac-site-isolation/TestExpectations:
* Tools/CISupport/build-webkit-org/config.json:
* Tools/CISupport/build-webkit-org/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/build-webkit-org/steps.py:
(RunAPITests):
* Tools/Scripts/webkitpy/layout_tests/controllers/single_test_runner.py:
(SingleTestRunner):
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
(parse_args):
(_set_up_derived_options):

Canonical link: https://commits.webkit.org/302686@main
